python/lsst/ts/MTMount/mtmount_csc.py

```diff
@@ -149,6 +179,81 @@ async def close_tasks(self):
         await super().close_tasks()
         await self.close_communication()
 
+    async def get_ccw_demand(self):
+        """Get CCW demand.
+
+        The method will get the position of the CCW and of the Rotator and
+        compute an optimum demand for the CCW. It takes into account how far
+        CCW and Rotator are from each other and also how far they both are
+        from the demand position.
+
+        Returns
+        -------
+        ccw_demand_position: `float`
+            CCW demand position (in degrees).
+        ccw_demand_velocity: `float`
+            CCW demand velocity (in degrees).
+
+        """
+
+        rot_data = await self.rotator.tel_Application.next(flush=True)
+        ccw_data = await self.mtmount.tel_Camera_Cable_Wrap.aget()
+
+        if self.last_rotator_position is not None:
+            dt = rot_data.private_sndStamp - self.last_rotator_time
+            # Estimate velocity if GAP between last rotator data and current
+            # data is smaller then MAX_ROTATOR_APPLICATION_GAP, else set
+            # velocity to zero.
+            rotator_velocity = (
+                (rot_data.Position - self.last_rotator_position) / dt
+                if dt < MAX_ROTATOR_APPLICATION_GAP
+                else 0.0
+            )
+        else:
+            rotator_velocity = 0.0
+
+        self.last_rotator_position = rot_data.Position
+        self.last_rotator_time = rot_data.private_sndStamp
+
+        ccw_angle = (ccw_data.CCW_Angle_1 + ccw_data.CCW_Angle_2) / 2.0
+
+        # distance between ccw and rotator position
+        distance_ccw_rot = ccw_angle - rot_data.Position
+
+        # distance between ccw and rotator demand
+        distance_ccw_rot_demand = ccw_angle - rot_data.Demand
+
+        # distance between rotator position and demand
+        distance_rot_demand = rot_data.Position - rot_data.Demand
+
+        if (
+            not self.catch_up_mode
+            and abs(distance_ccw_rot) < self.ccw_rot_sync_limit_slew
+        ) or (
+            abs(distance_ccw_rot_demand) < self.ccw_rot_sync_limit_goto
+            and abs(distance_ccw_rot) < self.ccw_rot_sync_limit_max / 2.0
+        ):
+            # CCW and Rotator synchronized or in the goto limit and CCW-Rotator
+            # Close enough. Follow demand.
+            return rot_data.Demand, 0.0
```

Comment:
Surely you should return the rotator demand velocity? You could compute that the same way you compute current velocity (and with the same validity criterion).

Reply:
Should you turn off catchup mode here? There is an `or` condition that will get you here even if we are presently in catchup mode.

Reply:
To the first question, the problem is that the demand position will change considerably when you are going from one target to another, so computing the demand velocity is non-trivial. I think we can update this once the rotator publishes that information. I'll add a `TODO`, but for now, I think this is simpler/more reliable.

To the second question, yes, that's a good idea.